### ISSUE TITLE
Issue loading meshes from URDF

### ISSUE THREAD

**Issue by davidgolub** [bulletphysics/bullet3#1933]
When loading a simple mesh from the following URDF, all of the color is gone without warning, even though the texture is there. But when I load the same mesh using p.createVisualShape(...), p.createMultibody(...) it loads fine. Any thoughts on what could be happening?  Thanks!

![image](
![screenshot from 2018-10-13 14-06-16](https://user-images.githubusercontent.com/12311502/46910022-c6a5e080-cef1-11e8-973c-29b09b42774f.png)
)

``

**Comment by erwincoumans** [comment]
Only 1 mesh/color per link.
You can change the color using
p.changeVisualShape(obUniqueId, linkIndex, rgbaColor=[r,g,b,a])
Otherwise, share the files, zipped.

**Comment by davidgolub** [comment]
Yes, there is only 1 mesh per a link. Will share the files right now.

**Comment by davidgolub** [comment]
[globe.tar.gz](https://github.com/bulletphysics/bullet3/files/2475846/globe.tar.gz)
The urdf file is in configuration_2017-07-18/globe.urdf, and the meshes are in the meshes folder. Note that you would have to change the absolute path of meshes in the urdf file to point to the proper meshes subdirectory on your machine

**Comment by davidgolub** [comment]
Thanks for your fast response. But in the URDF file we already converted all objects from DAE to OBJ format, and we're still facing the same issue.

**Comment by erwincoumans** [comment]
The color can be
A) assigned in the URDF file
~~~~~~~~~~~~
<link name ="rb1_marker">
    <visual>
      <origin xyz='-0.125441 0.088169 -0.129482' rpy='-1.936752 0.767602 1.007135'/>
      <geometry>
        <mesh filename="meshes/globe_sphere.dae"  scale="1 1 1" />
      </geometry>
       <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
    </visual>
  </link>
~~~~~~~~~~~~
B) assigned in PyBullet
~~~~~~~~~~~
p.changeVisualShape(obUniqueId, linkIndex, rgbaColor=[r,g,b,a])
~~~~~~~~~~~

Both versions work when I test the zip file.
Did you try method A and/or B?

Which of those don't work?
